app.js

```diff
@@ -154,31 +154,60 @@ function addEventListener(indicator, emitter, event, f) {
 }
 
 /*
-  Check the referenced database exists and that the credentials used have
+  Check the backup database exists and that the credentials used have
   visibility. Callback with a fatal error if there is a problem with the DB.
   @param {string} db - database object
   @param {function(err)} callback - error is undefined if DB exists
 */
-function proceedIfDbValid(db, callback) {
+function proceedIfBackupDbValid(db, callback) {
   db.service.headDatabase({ db: db.db }).then(() => callback()).catch(err => {
-    err = error.convertResponseError(err, function(err) {
-      if (err && err.status === 404) {
-        // Override the error type and mesasge for the DB not found case
-        var msg = `Database ${db.url}` +
-        `${db.db} does not exist. ` +
-        'Check the URL and database name have been specified correctly.';
-        var noDBErr = new Error(msg);
-        noDBErr.name = 'DatabaseNotFound';
-        return noDBErr;
-      } else {
-        // Delegate to the default error factory if it wasn't a 404
-        return error.convertResponseError(err);
-      }
-    });
+    err = error.convertResponseError(err, err => parseIfDbValidResponseError(db, err));
+    callback(err);
+  });
+}
+
+/*
+  Check the restore database exists, new and empty and that the credentials used have
+  visibility. Callback with a fatal error if there is a problem with the DB.
+  @param {string} db - database object
+  @param {function(err)} callback - error is undefined if DB exists, new and empty
+*/
+function proceedIfRestoreDbValid(db, callback) {
+  db.service.getDatabaseInformation({ db: db.db }).then(response => {
+    const { doc_count: docCount, doc_del_count: deletedDocCount } = response.result;
+    if (!db.db.startsWith("_") && (docCount !== 0 || deletedDocCount !== 0)) {
+      var notEmptyDBErr = new Error(`Database ${db.url} is not empty.`);
+      notEmptyDBErr.name = 'DatabaseNotEmpty';
```

Comment:
```suggestion
      notEmptyDBErr.name = 'RestoreDatabaseNotEmpty';
```

Reply:
Ok no problem, I'm fine with keeping it as-is.  I figured I'd bring it up if we think it's better to have it more specific especially if this function will only be used by restore functionality.